volumetric/3dVolMarchingCubes.cpp

```diff
@@ -113,102 +118,165 @@ int main( int argc, char** argv )
+  
+  
   bool parseOK=true;
   po::variables_map vm;
   try{
-    po::store(po::parse_command_line(argc, argv, general_opt), vm);  
+    po::store(po::parse_command_line(argc, argv, general_opt), vm);
   }catch(const std::exception& ex){
     parseOK=false;
     trace.info()<< "Error checking program options: "<< ex.what()<< std::endl;
   }
   po::notify(vm);
   if ( !parseOK || vm.count("help") || ( argc <= 1 ) )
-    {
-      std::cout << "Usage: " << argv[0]
-                << " [-i <fileName.vol>] [-t <threshold>] [-a <adjacency>] [-o <output.off>]" << std::endl
-                << "Outputs the isosurface of value <threshold> of the volume <fileName.vol>  as an OFF file <output.off>. The <adjacency> (0/1) allows to choose between interior (6,18) and exterior (18,6) adjacency." << std::endl
-                << general_opt << std::endl;
-      return 0;
-    }
+  {
+    std::cout << "Usage: " << argv[0]
+    << " [-i <fileName.vol>] [-t <threshold>] [-a <adjacency>] [-o <output.off>]" << std::endl
+    << "Outputs the isosurface of value <threshold> of the volume <fileName.vol>  as an OFF file <output.off>. The <adjacency> (0/1) allows to choose between interior (6,18) and exterior (18,6) adjacency." << std::endl
+    << general_opt << std::endl;
+    return 0;
+  }
   if ( ! vm.count("input") )
-    {
-      trace.error() << "The input file name was defined." << std::endl;
-      return 1;
-    }
+  {
+    trace.error() << "The input file name was defined." << std::endl;
+    return 1;
+  }
   std::string inputFilename = vm["input"].as<std::string>();
   unsigned int threshold = vm["threshold"].as<unsigned int>();
   bool intAdjacency = ( vm["adjacency"].as<unsigned int>() == 0 );
   std::string outputFilename = vm["output"].as<std::string>();
+  double noise ;
+  bool addNoise=false;
+  if (vm.count("noise") )
+  {
+    addNoise=true;
+    noise = vm["noise"].as<double>();
+  }
   //! [3dVolMarchingCubes-parseCommandLine]
-
+  
   //! [3dVolMarchingCubes-readVol]
   trace.beginBlock( "Reading vol file into an image." );
   typedef ImageSelector < Domain, int>::Type Image;
   Image image = VolReader<Image>::importVol(inputFilename);
-
+  
   typedef functors::SimpleThresholdForegroundPredicate<Image> ThresholdedImage;
   ThresholdedImage thresholdedImage( image, threshold );
-  // DigitalSet set3d (image.domain());
-  // SetFromImage<DigitalSet>::append<Image>(set3d, image,
-  //                                         threshold, 255 );
   trace.endBlock();
   //! [3dVolMarchingCubes-readVol]
-
+  
+  
   //! [3dVolMarchingCubes-KSpace]
   trace.beginBlock( "Construct the Khalimsky space from the image domain." );
   KSpace ks;
-  bool space_ok = ks.init( image.domain().lowerBound(),
-                           image.domain().upperBound(), true );
+  bool space_ok = ks.init( image.domain().lowerBound() - KSpace::Vector::diagonal(2),
```

Comment:
The larger domain size makes an error when doing the sMakeBoundary(). If removed it looks fine.